src/components/og-datepicker/og-datepicker.e2e.ts

```diff
@@ -0,0 +1,71 @@
+import { newE2EPage, E2EElement, E2EPage } from '@stencil/core/testing';
+
+describe('og-datepicker', () => {
+    let page: E2EPage;
+    let component: E2EElement;
+    let flyout: E2EElement;
+    let header: E2EElement;
+    let input: E2EElement;
+
+    // enum Month
+
+    beforeEach(async () => {
+        page = await newE2EPage();
+        // note: the testdate need to have a clickable tomorrow
+        await page.setContent(`<og-datepicker value="30-05-2019" format="DD-MM-YYYY" first-day-of-week="1"></og-datepicker>`);
+
+        component = await page.find('og-datepicker');
+        flyout = await page.find('og-datepicker >>> og-calendar');
+        header = await page.find('og-datepicker >>> .og-datepicker__header');
+        input = await page.find('og-datepicker >>> .og-datepicker__input');
+        await page.waitForChanges();
+    });
+
+    it('renders', async () => {
+        expect(component).toHaveClass('hydrated');
+        expect(flyout).toHaveClass('hydrated');
+    });
+
+    it('is hidden by default', async () => {
+        expect(flyout).not.toHaveClass(
+            "og-datepicker__flyout__calendar--visible"
+        );
+    });
+
+    it('is visible when called and hidden after closing', async () => {
+        await header.click();
+        await page.waitForChanges();
+
+        expect(flyout).toHaveClass(
+            'og-datepicker__flyout__calendar--visible'
+        );
+
+        await header.click();
+        await page.waitForChanges();
+
+        expect(flyout).not.toHaveClass(
+            'og-datepicker__flyout__calendar--visible'
+        );
+    });
+
+    it('displays the date in the right format', async () => {
+        expect(await input.getProperty('value')).toEqualText('30-05-2019');
+    });
+
+    // todo: check this
```

Comment:
did someone had a look at this?

Reply:
https://github.com/ionic-team/stencil/issues/1530